executor/adapter.go

```diff
@@ -463,3 +464,50 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx sessionctx.Context, p plannerco
 		return false, nil
 	}
 }
+
+// execCtxWatcher used to watch the context specified in Open and
+// stop the wrapped executor when this context is cancelled.
+type execCtxWatcher struct {
```

Comment:
Will this affect the check of handleNoDelayExecutor?

Reply:
Yes, you are right, I will fix it.

Reply:
fixed